joint_pid_controller_node:
  ros__parameters:
    control_loop_rate: 50.0 # Hz

    feedback_joint_states_topic: "/joint_states"

    # PID and controller settings for joint1
    joint1:
      target_joint_name: "joint1"
      effort_command_topic: "/joint1_effort_controller/commands"
      target_angle_topic: "/joint1_target_angle"
      pid:
        kp: 0.0
        ki: 0.0
        kd: 0.0
        integral_clamp_min: -20.0
        integral_clamp_max: 20.0
      output_limits:
        min: -80.0
        max: 80.0

    # PID and controller settings for joint2
    joint2:
      target_joint_name: "joint2"
      effort_command_topic: "/joint2_effort_controller/commands"
      target_angle_topic: "/joint2_target_angle"
      pid:
        kp: 2300.0
        ki: 300.0
        kd: 400.0
        integral_clamp_min: -500.0
        integral_clamp_max: 500.0
      output_limits:
        min: -250.0
        max: 250.0